server/konnectors/maif.js

```diff
@@ -0,0 +1,302 @@
+    });
+  });
+}
+
+
+/**
+* function called after getToken
+* sends get request with token to get JSON data in return
+*/
+function getData(token, callback) {
+  MaifUser.first((err, maifuser) => {
+    const options = {
+      url: infoUrl,
+      jar: true,
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${token}`,
+      }
+    };
+
+    request(options, (err, response, body) => {
+      try {
+        JSON.parse(body);
+      } catch (e) {
+        err = 'error';
+      }
+      if (err != null) {
+        sendNotification('data retrieved failed', 'konnectors/konnector/maif');
+        callback(err);
+      } else {
+        moment.locale('fr');
+        const importDate = moment().format('LLL');
+        const payload = { profile: JSON.parse(body), date: importDate };
+
+        maifuser.updateAttributes(payload, (err) => { // mise à jour du maifuser en base en insérant le token
+          sendNotification('data retrieved', 'mes-infos-maif');
+          callback(err);
+        });
+      }
+    }, false);
+  });
+}
+
+/**
+* refreshToken function
+* called at each scheduled import (every hour/day/week/month)
+* get new token and refresh token
+* call getToken with token and refresh token
+*/
+function refreshToken(requiredFields, entries, data, next) {
+  MaifUser.first((err, maifUser) => {
+    let tokenValid = true;
+    if (maifUser !== undefined) {
+      const token = maifUser.password;
+      if (token !== undefined) {
+        const options = {
+          url: `${connectUrl}/token`,
+          jar: true,
+          method: 'POST',
+          headers: {
+            Authorization: `Basic ${b64Client}`,
+          },
+          form: {
+            grant_type: 'refresh_token',
+            refresh_token: token,
+          }
+        };
+        request(options, (err, response, body) => {
+          try {
+            JSON.parse(body);
+          } catch (e) {
+            err = 'parsing error';
+          }
+          if (err != null) { // refresh token not valid anymore
+            sendNotification('refresh token not valid', 'konnectors/konnector/maif');
+            next(err);
+          } else {
+            const jsonToken = JSON.parse(body);
+            getToken(jsonToken.token, jsonToken.token_refresh, next);
+          }
+        }, false);
+      } else {
+        tokenValid = false;
+      }
+    } else {
+      tokenValid = false;
+    }
+    if (!tokenValid) {
+      // Maybe we have no token yet !
+      getCode(requiredFields, (err) => {
+        if (err) {
+          sendNotification('refresh token not valid', 'konnectors/konnector/maif');
+          connecteur.logger.error(err);
+          next('token not found');
+        } else {
+          next();
+        }
+      });
+    }
+  });
+}
+
+/**
+* Display a notification in Cozy
+* code : code of the message to send
+* appToOpen : Link to the app that will be opened on notification's click
+*/
+function sendNotification(code, appToOpen) {
```

Comment:
I would expect that clicking on 'Connect' opens a new tab. can you add `target="_blank"` here : https://github.com/cozy-labs/konnectors/blob/master/client/app/views/konnector.coffee#L330